src/common/config.cc

```diff
@@ -474,6 +475,30 @@ void md_config_t::parse_env(ConfigValues& values,
       _set_val(values, tracker, dir, *o, CONF_ENV, &err);
     }
   }
+  const char *pod_req = getenv("POD_MEMORY_REQUEST");
+  if (pod_req) {
+    switch (entity_type) {
+    case CEPH_ENTITY_TYPE_OSD:
+      _set_val(values, tracker, pod_req,
+	       *find_option("osd_memory_target"),
+	       CONF_ENV, nullptr);
+      break;
+    }
+  }
+  uint64_t cgroup_limit = 0;
+  if (get_cgroup_memory_limit(&cgroup_limit) == 0) {
+    // this is an "environment" property in the general sense, so
+    // populate the option at level CEPH_DEFAULT.  that way a user can
+    // override with the normal (e.g., mon) config.
+    cgroup_limit *= .8;
```

Comment:
I'd suggest a config option for this `.8`.

Reply:
Yeah... the problem is that snowballs pretty quickly.  That config option might come from the mon, or a config file, or via admin socket 'config set', which means we don't have the value now. That means we need to move this logic somewhere much later.  Let me think about where that would be.

Reply:
Ok, I think teh way to do this is downstream of the core config parsing.  Take a look...